editor/plugins/audio_stream_editor_plugin.cpp

```diff
@@ -125,10 +132,13 @@ void AudioStreamEditor::_stop() {
 
 void AudioStreamEditor::_on_finished() {
 	_play_button->set_icon(get_theme_icon("MainPlay", "EditorIcons"));
-	if (_current == _player->get_stream()->get_length()) {
-		_current = 0;
-		_indicator->update();
-	}
+
+	// TODO: _current is always less than _player->get_playback_position() when the
+	// signal is emitted. Turn off processing here, and reset the indicator.
```

Comment:
There's a TODO but it's not clear what needs to be fixed and how. Is the TODO created by your changes, or is it a pre-existing issue?

Reply:
It's created by my changes.

`_current` isn't updated properly before the signal is emitted, so `_current` is always less than `_player->get_playback_position()`. That's why this conditional always fails:
https://github.com/godotengine/godot/blob/6010c49424b4e5a1110b6c5a17ef2fdbfc398067/editor/plugins/audio_stream_editor_plugin.cpp#L128-L131

Instead, I just decided to reset `_current` in `_play()`, rather than dealing with the underlying race condition.

I'll change the comment.